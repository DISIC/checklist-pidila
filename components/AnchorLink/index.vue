<template>
    <a tabindex="0" @click="scrollTo(anchor)" @keyup.enter="scrollTo(anchor)" :class="cssClass">
      {{message}}
    </a>
</template>

<script>
  export default {
    name: 'AnchorLink',
    props: {
      anchor: {
        type: String,
        default: ''
      },
      cssClass: {
        type: String,
        default: ''
      },
      message: {
        type: String,
        default: ''
      }
    },
    methods: {
      scrollTo: function (anchor) {
        this.$router.scrollTo(anchor)
      }
    }
  }
</script>
