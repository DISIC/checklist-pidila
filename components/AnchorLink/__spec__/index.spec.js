import {shallowMount} from '@vue/test-utils'
import AnchorLink from '../index.vue'

describe('components | AnchorLink', () => {
  const props = {
    anchor: 'main',
    message: 'Aller au contenu',
    cssClass: 'sr-only sr-only-focusable'
  }
  const $router = {}

  beforeEach(() => {
    $router.scrollTo = jest.fn()
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(AnchorLink, {mocks: {$router}})

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should display the expected message', () => {
    // when
    const wrapper = shallowMount(AnchorLink, {propsData: props, mocks: {$router}})

    // then
    expect(wrapper.find('a').element.textContent.trim()).toBe('Aller au contenu')
  })

  it('should contain the right css props class', () => {
    // when
    const wrapper = shallowMount(AnchorLink, {propsData: props, mocks: {$router}})

    // then
    expect(wrapper.find('a').element.className).toBe('sr-only sr-only-focusable')
  })

  it('should call the scrollTo function from router instance with anchor when clicked', () => {
    // given
    const wrapper = shallowMount(AnchorLink, {propsData: props, mocks: {$router}})

    // when
    wrapper.find('a').trigger('click')

    // then
    expect($router.scrollTo).toHaveBeenCalledWith('main')
  })
})
