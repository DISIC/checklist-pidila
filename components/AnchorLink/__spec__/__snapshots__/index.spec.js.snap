// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | AnchorLink should match snapshot 1`] = `
<a
  class=""
  tabindex="0"
>
  
  

</a>
`;
